soc/realtek/ameba/common/os_wrapper/include/dlist.h

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2024 Realtek Semiconductor Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __LIST_H
+#define __LIST_H
+
+/* This file is from Linux Kernel (include/linux/list.h)
+ * and modified by simply removing hardware prefetching of list items.
+ * Here by copyright, credits attributed to wherever they belong.
+ * Kulesh Shanmugasundaram (kulesh [squiggly] isis.poly.edu)
+ */
```

Comment:
https://github.com/torvalds/linux/blob/6c52d4da1c742cd01a797a4d0a2d3c5a60dc9bfe/include/linux/list.h#L1

You can't just take GPL 2.0-licensed code and re-license it as Apache 2.0 ...

Also, why can't you use the existing list API in Zephyr?

Reply:
thanks, I have change to use the  existing list API in Zephyr.